### PR TITLE
Add basic slack CLI support

### PR DESCRIPTION
**Pull request by WilliamBergamin**
The goal of this PR is to introduce **basic** support for slack CLI in Bolt for python. This is accomplished by implementing the `get-hooks`, `get-manifest` and `start` interface. Support for `default` and `message-boundaries` communication protocol was also implemented in order to leverage the full capabilities of the slack CLI

#### Testing

1. `slack create -t https://github.com/slack-samples/bolt-python-starter-template -b cli-support`
2. (Optional) In your project set up a venv with python -m venv .venv and source .venv/bin/activate
3. `pip install -r requirements.txt`
4. `pip uninstall slack-bolt`
5. Pull the latest PR branch of bolt python to your local
6. In the Bolt-Python project run `scripts/build_pypi_package.sh` this will generate a `.whl` in the `dist/` folder
8. Back in your project run `pip install global/path/to/bolt-python/dist/something.whl`
9. Add the following fields in the `manifest.json`
    - `"outgoing_domains": []`
    - `"settings"."function_runtime": "remote"`
    - `"settings"."org_deploy_enabled": true`
10. Now you should be able to run the following 
    - `slack manifest`
    - `slack run`

#### Feedback

- Code implementation, what could be improved are there any technical issues with this approach
- I was able to manually test `slack run` for async apps but had issues implementing scenario tests for them, any ideas for this can help 
- User experience, are error clear and handled nicely, does this integrate well with the existing capabilities of the CLI
- Are there any higher level solutions I might have missed that are worth discussing
- Any concerns you may have on this, this addition has any moving components I very well may have missed something

### Category

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [ ] Adapters in `slack_bolt.adapter`
* [ ] Document pages under `/docs`
* [x] Others

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

**Comment by filmaj** [comment]
@WilliamBergamin the example template listed in the testing instructions above doesn't include a Custom Function, right?

**Comment by WilliamBergamin** [comment]
> the example template listed in the testing instructions above doesn't include a Custom Function, right?

No it does not include a custom function

**Comment by seratch** [comment]
@WilliamBergamin Either way, developers need to install required packges (at least slack-bolt and its underlying dependecny slack-sdk
). If our CLI mechanism has a limitation that developers cannot smoothly import multiple PyPI packages, 
>Should the slack-cli-hooks project import slack-bolt for the developer?

I am down to this idea. With this approach, the slack-cli-hooks can resolves slack-bolt and slack-sdk automatically. Developers do not need to know what packages are required to spin up their apps.

File: slack_bolt/cli/get_hooks.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import json
+from .protocol import Protocol, MessageBoundaryProtocol, DefaultProtocol, protocol_factory
+from slack_bolt.cli import get_manifest, start
+
+PROTOCOL: Protocol
+
+hooks_payload = {
+    "hooks": {
+        "get-manifest": f"python -m {get_manifest.__name__}",
+        "start": f"python -m {start.__name__}",
```
Comment (mwbrooks) [inline slack_bolt/cli/get_hooks.py:11]: Are there any concerns around using the `python` runtime? For example, my macOS system doesn't have a `python` binary installed and instead uses `python3`.
image

Comment (WilliamBergamin) [reply slack_bolt/cli/get_hooks.py:11]: Using `python3` instead of `python` seem to be the safer option

**Pull request closed without merging**